fluent_pose_synthesis/tests/test_models.py

```diff
@@ -0,0 +1,55 @@
+import os
+import torch
+import pytest
+from fluent_pose_synthesis.network.models import SignLanguagePoseDiffusion
+
+
+def get_dummy_batch(batch_size=2, seq_len=50, keypoints=178, dims=3):
+    """
+    Returns a minimal dummy batch dict like the example:
+      - data: (B, seq_len, 1, keypoints, dims)
+      - conditions.input_sequence: same shape
+      - t: zeros (B,)
+    """
+    data = torch.ones(batch_size, seq_len, 1, keypoints, dims)
+    conditions = {"input_sequence": torch.ones(batch_size, seq_len, 1, keypoints, dims)}
+    t = torch.zeros(batch_size, dtype=torch.long)
+    
+    return data, conditions, t
+
+
+@pytest.mark.parametrize("arch", ["trans_enc", "trans_dec", "gru"])
+@pytest.mark.parametrize("seq_len", [50, 100])
+@pytest.mark.parametrize("batch_size", [2, 4, 16])
+def test_base_output_shape(arch, seq_len, batch_size):
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
for test, use `cpu`, no need to init cuda for one forward pass (can make model very small, `ff_size=256, num_layers=2`)